Meadowlark-js/packages/meadowlark-authz-server/test/handler/GetClients.test.ts

```diff
@@ -0,0 +1,313 @@
+      });
+
+      mockMeadowlarkCore = jest.spyOn(JwtAction, 'verifyJwt').mockReturnValue({
+        ...newJwtStatus(),
+        isValid: true,
+        roles: ['admin'],
+      });
+
+      // Act
+      response = await getClients(authorizationRequest);
+    });
+
+    afterAll(() => {
+      mockAuthorizationStore.mockRestore();
+      mockMeadowlarkCore.mockRestore();
+    });
+
+    it('returns status 200', () => {
+      expect(response.statusCode).toEqual(200);
+    });
+
+    it('returns a proper message body', () => {
+      expect(response.body).toMatchInlineSnapshot(`"[]"`);
+    });
+  });
+
+  describe('given missing authorization token', () => {
+    let response: AuthorizationResponse;
+    let mockMeadowlarkCore: any;
+
+    beforeAll(async () => {
+      mockMeadowlarkCore = jest.spyOn(JwtAction, 'verifyJwt').mockReturnValue({
+        ...newJwtStatus(),
+        isMissing: true,
+      });
+
+      // Act
+      response = await getClients(authorizationRequest);
+    });
+
+    afterAll(() => {
+      mockMeadowlarkCore.mockRestore();
+    });
+
+    it('returns error response', () => {
+      expect(response).toMatchInlineSnapshot(`
+              {
+                "body": "{ "error": "invalid_client", "error_description": "Authorization token not provided" }",
+                "headers": {
+                  "WWW-Authenticate": "Bearer",
+                },
+                "statusCode": 401,
+              }
+          `);
+    });
+  });
+
+  describe('given expired authorization token', () => {
+    let response: AuthorizationResponse;
+    let mockMeadowlarkCore: any;
+
+    beforeAll(async () => {
+      mockMeadowlarkCore = jest.spyOn(JwtAction, 'verifyJwt').mockReturnValue({
+        ...newJwtStatus(),
+        isMissing: false,
+        isExpired: true,
+      });
+
+      // Act
+      response = await getClients(authorizationRequest);
+    });
+
+    afterAll(() => {
+      mockMeadowlarkCore.mockRestore();
+    });
+
+    it('returns error response', () => {
+      expect(response).toMatchInlineSnapshot(`
+              {
+                "body": "{ "error": "invalid_token", "error_description": "Token is expired" }",
+                "headers": {
+                  "WWW-Authenticate": "Bearer",
+                },
+                "statusCode": 401,
+              }
+          `);
+    });
+  });
+
+  describe('given non-admin authorization token', () => {
+    let response: AuthorizationResponse;
+    let mockMeadowlarkCore: any;
+
+    beforeAll(async () => {
+      mockMeadowlarkCore = jest.spyOn(JwtAction, 'verifyJwt').mockReturnValue({
+        ...newJwtStatus(),
+        isMissing: false,
+        isExpired: false,
+        isValid: true,
+        roles: ['vendor'],
+      });
+
+      // Act
+      response = await getClients(authorizationRequest);
+    });
+
+    afterAll(() => {
+      mockMeadowlarkCore.mockRestore();
+    });
+
+    it('returns error response', () => {
```

Comment:
I need to look at this more closely. Since this is an authorization problem, an empty body does make sense. But we should probably be looking at the status code too.

Reply:
```suggestion
    it('returns status code 403', () => {
      expect(response.statusCode).toBe(403);
    });

    it('returns empty message body', () => {
```